src/formats/ecl.rs

```diff
@@ -1082,7 +1103,7 @@ impl InstrFormat for OldeEclHooks {
     fn read_instr(&self, f: &mut BinReader, emitter: &dyn Emitter) -> ReadResult<ReadInstr> {
         let time = f.read_i32()?;
         let opcode = f.read_u16()?;
-        let size = f.read_u16()? as usize;
+        let size = f.read_i16()? as usize;
```

Comment:
it seems to me this is just begging for an Out of Memory error without providing any advantages

Reply:
The field *is* signed though, read with MOVSX. If a size is greater than 0x7FFF, it'll index backwards (and probably explode)

Reply:
It doesn't matter whether it's signed in game because it gets cast immediately to an unsigned type.  So unless you plan to add range checking (which you certainly can do), you're literally just trading one incorrect behavior for another...

Reply:
Range checking would definitely be ideal, though IDK how you'd want the error to be formatted.

Reply:
Seems similar to the check in anm